core/src/main/scala/org/typelevel/paiges/Doc.scala

```diff
@@ -570,8 +570,11 @@ object Doc {
   /**
    * Invariant: generate is referentially transparent
    */
-  private[paiges] case class LazyDoc(private val generate: () => Doc) extends Doc {
+  private[paiges] class LazyDoc(private val generate: () => Doc) extends Doc {
     lazy val evaluated: Doc = generate()
+    override def canEqual(that: Any): Boolean = false
+    override def productElement(n: Int): Any = None
```

Comment:
This should return either generate or evaluated for n == 0 and throw in the other cases. I’d say generate.